src/UserManager.ts

```diff
@@ -113,24 +112,23 @@ export class UserManager {
         return user;
     }
 
-    public async signinPopup(): Promise<User> {
+    public async signinPopup(params?: PopupWindowParams): Promise<User> {
         const url = this.settings.popup_redirect_uri || this.settings.redirect_uri;
         if (!url) {
             Log.error("UserManager.signinPopup: No popup_redirect_uri or redirect_uri configured");
             throw new Error("No popup_redirect_uri or redirect_uri configured");
         }
 
-        const args = {
-            request_type: "si:p",
-            redirect_uri: url,
-            display: "popup"
-        };
-        const user = await this._signin(args, this._popupNavigator, {
-            startUrl: url,
+        const handle = await this._popupNavigator.prepare({
             popupWindowFeatures: this.settings.popupWindowFeatures,
             popupWindowTarget: this.settings.popupWindowTarget,
-            redirectMethod: this.settings.redirectMethod
+            ...params
         });
+        const user = await this._signin({
+            request_type: "si:p",
+            redirect_uri: url,
+            display: "popup"
```

Comment:
hmm original args/params was passed into _signin and not navigatorParams...
See  https://github.com/IdentityModel/oidc-client-js/blob/d700897e0a9e5eac060b12a687d55606535ca5e9/src/UserManager.js#L127

Reply:
Could you clarify a bit? In the linked line, I've combined the 2nd and 3rd args to create the `handle` instance that I pass to `_signin`. The object passed to `_signin` is the same object that was previously declared on its own a few lines before.

Reply:
in order to restore original functionality (https://github.com/IdentityModel/oidc-client-js/blob/dev/src/UserManager.js#L127), how it was:
public signinRedirect(**args**) -> this._signinStart(**args**,... -> this.createSigninRequest(**args**)

https://github.com/pamapa/oidc-client-ts/blob/456addb3ea4e247a5e43f1cf813dd7da04fc3bf3/src/UserManager.ts#L93:
You are now passing **by default** arguments to the navigator and _only_ extraQueryParams, extraTokenParams. To restore functionality it might be better to have the sigin arguments the default and handle 

Your code:
```
signinRedirect({
        extraQueryParams,
        extraTokenParams,
        **...navigatorParams**
    }: ExtraSigninRequestArgs & RedirectParams = {}
```
-> would restore not mean? With this approach its possible to inject more as sigin request args... even if we do not explicitly declare the types...
```
signinRedirect({
    redirectMethod?: "replace" | "assign",
    **...args**
}: ExtraSigninRequestArgs & RedirectParams = {}
```

-> ~~Maybe its better to not mix up sigin arguments with navigation parameters on the API at all. As such we could avoid what the _default_ is. This makes the API more difficult to use.~~
```
**not** signinRedirect(args?: ExtraSigninRequestArgs = {}, navigatorParams?: RedirectParams = {})
```

Reply:
Ok - so as far as passing extra navigator params "by default", I don't see it as an issue because I do explicitly type the extent of what can be passed to those methods, but I can reverse them you really think it improves readability.

On being able to inject more signin request args, I really do think that the ones chosen (`extraQueryParams` and `extraTokenParams`) are all that are necessary, adding more would overcomplicate the function signature. The code I have now causes a type error when passing additional properties.
![image](https://user-images.githubusercontent.com/4993980/134920825-b4111e34-75c1-42c4-ab43-00ddbc863de8.png)


I considered splitting up the objects like you say here
> -> Maybe its better to not mix up sigin arguments with navigation parameters on the API at all. As such we could avoid what the default is. This makes the API more difficult to use.
> <pre><code>**not** signinRedirect(args?: ExtraSigninRequestArgs = {}, navigatorParams?: RedirectParams = {})</code></pre>

and I still like the idea, but it's definitely easier to work with a function that only takes 1 argument vs 2.

Reply:
please reverse them, lets keep args the default (even if we have only extraQueryParams and extraTokenParams) and navigiation params shall be the special extra handled ones

Reply:
I added a commit testing what two parameters would look like, and I don't hate it. I still feel like customizing the request args with `extraQueryParams` is the exceptional case. Your suggestion is reasonable though, I'll reverse the grouping and am happy to leave it at that